pkg/solana/codec/testutils.go

```diff
@@ -0,0 +1,267 @@
+package codec
```

Comment:
I would put this in a different package like testutils under codec.  I don't like when test functions leak to the implementation.

Reply:
If I put it under `codec`, it creates a circular dependency. I had thought about putting it in `pkg` or `internal`, but that felt really messy. Maybe `internal/codec/testutils`?

Reply:
I removed the function that would create a circular dependency and moved the contents to `testutils` in `codec`. I also moved json to files and embedded them.